lib/xbr/buyer.js

```diff
@@ -0,0 +1,118 @@
+var cbor = require('cbor');
+var nacl = require('tweetnacl');
+var eth_accounts = require("web3-eth-accounts");
+var eth_util = require("ethereumjs-util");
+
+
+var SimpleBuyer = function (buyerKey, maxPrice) {
+    this._running = false;
+    this._session = null;
+    this._channel = null;
+    this._balance = null;
+    this._keys = {};
+    this._maxPrice = maxPrice;
+
+    var account = new eth_accounts.Accounts().privateKeyToAccount(buyerKey);
+    this._addr = eth_util.toBuffer(account.address);
+
+    this._keyPair = nacl.box.keyPair();
+};
+
+SimpleBuyer.prototype.start = function(session, consumerID, on_success, on_failure) {
```

Comment:
@oberstet Need your input on this API signature. Currently it takes two callbacks `on_success` and `on_failure` as parameters because `start()` makes a call to `session.call()` and returns instantly, hence we can't really return the result as we do in the Python implementation.

Are you fine with that approach or do we want to use something different here (and in other places below).

Reply:
`start` should return a proper promise (not take callback args), and then you can fire the promise with success or error later on ..

Reply:
rgd new release: sure!